lib/active_record_host_pool/connection_adapter_mixin.rb

```diff
@@ -32,27 +32,27 @@ def _host_pool_current_database=(database)
     end
 
     def execute_with_switching(*args)
-      if _host_pool_current_database && ! @_no_switch
+      if _host_pool_current_database && !_no_switch
         _switch_connection
       end
       execute_without_switching(*args)
     end
 
     def drop_database_with_no_switching(*args)
       begin
-        @_no_switch = true
+        _no_switch = true
         drop_database_without_no_switching(*args)
       ensure
-        @_no_switch = false
+        _no_switch = false
```

Comment:
this is not working since it's assigning a local var ... so missing tests ...
extract `without_switching` method ?